### ISSUE TITLE
Results cannot be reproduced

### ISSUE THREAD

**Issue by ijwky** [KindXiaoming/pykan#523]
I think I have controlled all the randomness, but the result still cannot be replicated with all the same random seeds. 

torch.manual_seed(6)
random.seed(6)
np.random.seed(6)
if torch.cuda.is_available():
    torch.cuda.manual_seed_all(6)

from kan import *
torch.set_default_dtype(torch.float64)

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
print(device)

# create a KAN: [input_num, hidder_neurons_num, output_num] 
#1D inputs, 1D output, and 5 hidden neurons. cubic spline (k=3), 5 grid intervals (grid=5).
kan_model = KAN(width = [1,5,1], grid = 3, k = 3, seed = 6, device = device)

k = 0.5  # True coefficient for Y
C0 = 0.1  # True constant term
f = lambda x: k * x + C0  #  f(Y) -> C

n = 100  # Number of observations
dataset = create_dataset(f, n_var=1, train_num = n, ranges=[0, 1], seed = 6)

Y = dataset['train_input']  
C = dataset['train_label']  

# plot KAN at initialization
kan_model(dataset['train_input']);
kan_model.plot()

# train the model
kan_model.fit(dataset, opt="LBFGS", steps=50, lamb=0.001);

kan_model.plot()

kan_model = kan_model.prune()
kan_model.plot()

kan_model.fit(dataset, opt="LBFGS", steps=50);

kan_model.symbolic_formula()

kan_model = kan_model.refine(10)

kan_model.fit(dataset, opt="LBFGS", steps=50);

mode = "auto" # "manual"

if mode == "manual":
    # manual mode
    kan_model.fix_symbolic(0,0,0,'sin');
    kan_model.fix_symbolic(0,1,0,'x^2');
    kan_model.fix_symbolic(1,0,0,'exp');
elif mode == "auto":
    # automatic mode
    lib = ['x','x^2','x^3','x^4','exp','log','sqrt']
    kan_model.auto_symbolic(lib=lib)

kan_model.fit(dataset, opt="LBFGS", steps=50);

from kan.utils import ex_round
ex_round(kan_model.symbolic_formula()[0][0],4)

C_pred_kan = kan_model(dataset['train_input'])
C_pred_kan_np = C_pred_kan.detach().cpu().numpy()
mse_kan = mean_squared_error(C, C_pred_kan_np)
print(mse_kan)

**Comment by KindXiaoming** [comment]
LBFGS may use non-deterministic operations, could you try adding `torch.use_deterministic_algorithms(True)`? Also, an ablation test would be trying opt = "adam" instead of "LBFGS" since Adam doesn't use non-deterministic operations.

**Comment by ijwky** [comment]
@KindXiaoming This really works! Thank you very much!
